### PR TITLE
Add documentation for override Swagger docs

### PR DESCRIPTION
**Pull request by Tim59**
Hello, 
I see some issues(questions) about the modification of the swagger documentation, i did a pull request

### CODE REVIEW

File: core/swagger-decorator.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
+# Override Swagger documentation
```
Comment (dunglas) [inline core/swagger-decorator.md]: I'll suggest to add a new documentation entry for Swagger.

I'll call the file `swagger.md` and the title `Swagger Support`.

File: core/swagger-decorator.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+
+Symfony allow to [decorates services](https://symfony.com/doc/current/service_container/service_decoration.html), here we need to decorate the
```
Comment (dunglas) [inline core/swagger-decorator.md]: `allows to decorate`

File: core/swagger-decorator.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+1. declare the decorator and class
+
+ ```yaml
+ # app/config/services.yml
+ 
+ services:
+      app.swagger.swagger_decorator:
+         decorates: api_platform.swagger.normalizer.documentation
+         class: AppBundle\Swagger\SwaggerDecorator
```
Comment (dunglas) [inline core/swagger-decorator.md]: Can you put the class name into single quotes (sometimes can cause problem to not escape it)

File: core/swagger-decorator.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+         arguments: ['@app.swagger.swagger_decorator.inner']
+ ```
+
+```php
+<?php
```
Comment (dunglas) [inline core/swagger-decorator.md]: Can you add a comment with the path of the `.php` file?

File: core/swagger-decorator.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+namespace AppBundle\Swagger;
+
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+
+/**
+ * Class SwaggerDecorator.
```
Comment (dunglas) [inline core/swagger-decorator.md]: Useless, can be remove (or replaced by an explicit description).

File: core/swagger-decorator.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+ */
+class SwaggerDecorator implements NormalizerInterface
```
Comment (dunglas) [inline core/swagger-decorator.md]: `final` (according to our CS)

File: core/swagger-decorator.md [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+{
+    /**
+     * @var NormalizerInterface
```
Comment (dunglas) [inline core/swagger-decorator.md]: Useless, will be inferred by the IDE, can be removed.

File: core/swagger-decorator.md [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+    public function supportsNormalization($data, $format = null)
+    {
+        return $this->decorated->supportsNormalization($data, $format);
+    }
+}
+```
+
+2. In Swagger decorator you can acces to the docs
+
+the variable `docs` is an array compose by paths :
```
Comment (dunglas) [inline core/swagger-decorator.md]: no space before the colon

File: core/swagger-decorator.md [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+
+`$docs['paths']['{path}']['{method}']['parameters']`
+
+for example if you want to remove all references to your path ’/foos’ in method GET in the swagger you can make in the normalize function
```
Comment (dunglas) [inline core/swagger-decorator.md]: you can do it in the `normalize` method

**Comment by dunglas** [comment]
Thank you for handling this @Tim59, it's definitely useful! What do you think about introducing a real life example (like changing the path as you do in the end of this article) instead of an abstract one? It will make it easier to understand for newcomers.

**Comment by Tim59** [comment]
@dunglas maybe something like that ?

File: core/swagger.md [hunk 10]
```diff
@@ -0,0 +1,71 @@
+# Swagger support
+
+## Override Swagger documentation
+
+Symfony allows to [decorate services](https://symfony.com/doc/current/service_container/service_decoration.html), here we need to decorate the
```
Comment (dunglas) [inline core/swagger.md]: `the` should be removed, blank lines can be removed too

File: core/swagger.md [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+
+`api_platform.swagger.normalizer.documentation`
+
+### Example
+
+In the following example, we will see how to override Swagger title and add a custom filter in `/foos` path only for the `GET` method
```
Comment (dunglas) [inline core/swagger.md]: `the title of the Swagger documentation` [...] `for the `GET` operation of the `/foos` path`.

File: core/swagger.md [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+
+// src\AppBundle\Swagger\SwaggerDecorator.php
+
+namespace AppBundle\Swagger;
+
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+
+final class SwaggerDecorator implements NormalizerInterface
+{
+
```
Comment (dunglas) [inline core/swagger.md]: Extra blank line to remove

File: core/swagger.md [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+        $customDefinition = [
+            'name' => 'fields',
+            'definition' => 'Fields to remove of the outpout',
+            'default' => 'id',
+            'in' => 'query',
+        ];
+
+		
+		// e.g add a custom parameter
```
Comment (dunglas) [inline core/swagger.md]: Can you remove one level of indentation? And `e.g.` (nitpicking)

File: core/swagger.md [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+    }
+
+    public function supportsNormalization($data, $format = null)
+    {
+        return $this->decorated->supportsNormalization($data, $format);
+    }
+}
+```
+
+
```
Comment (dunglas) [inline core/swagger.md]: Extra blank line to remove

File: core/swagger.md [hunk 15]
```diff
@@ -0,0 +1,71 @@
+# Swagger support
```
Comment (dunglas) [inline core/swagger.md]: `Swagger Support` (title case)

**Comment by Tim59** [comment]
@dunglas , it's update :)

Do you think a new section with ApiNormaliser decoration in other PR may be interesting ?

**Comment by dunglas** [comment]
Thank you @Tim59

**Comment by Tim59** [comment]
@dunglas you're welcome, about the ApiNormaliser decorator do you have advice (Title section, new file?)

**Comment by Tim59** [comment]
Yes i'm agree with that, I will make a PR with what i know because i'm not expert in that section :)
Thx for your help

**Pull request merged**